marc_to_solr/lib/traject_config.rb

```diff
@@ -736,16 +736,18 @@
   accumulator.replace(codes.flatten)
 end
 
+# # import PrimaryLanguage method from ./primary_laguage.rb
 to_field 'language_iana_s', extract_marc('008[35-37]:041a:041d') do |_record, accumulator|
-  codes = accumulator.compact.map { |c| c.length == 3 ? c : c.scan(/.{1,3}/) }.flatten.uniq
-  codes = codes.reject { |c| ISO_639.find(c).nil? }.map do |c|
-    if ISO_639.find(c).alpha2.empty?
-      c
+  codes = accumulator.compact.map { |m| m.length == 3 ? m : m.scan(/.{1,3}/) }.flatten.uniq
+  codes_iso_639 = codes.reject { |m| ISO_639.find(m).nil? }.map do |m|
+    if ISO_639.find(m).alpha2.empty? || ["zxx", "mul", "sgn", "und", "|||"].include?(m)
```

Comment:
Where do there codes in the include list come from?  It would be nice to understand why you are ignoring them.  Maybe it is as simple as naming them?

Reply:
@carolyncole good point. I looked with @mzelesky case by case and listed the values that should be excluded based on the section `35-37 - Language` in https://www.loc.gov/marc/bibliographic/bd008a.html. I will add a comment. Some of them are excluded from the ISO-639 gem but some are not. This is why I listed all of them so that I make sure I ignore the invalid ones.